block-storage/src/main/scala/coop/rchain/blockstorage/FileLMDBIndexBlockStore.scala

```diff
@@ -0,0 +1,255 @@
+package coop.rchain.blockstorage
+
+import java.io.RandomAccessFile
+import java.nio.ByteBuffer
+import java.nio.file.{Files, Path}
+import java.util.stream.Collectors
+
+import cats.Monad
+import cats.effect.{Concurrent, ExitCase, Sync}
+import cats.implicits._
+import cats.effect.concurrent.{Ref, Semaphore}
+import com.google.protobuf.ByteString
+import coop.rchain.blockstorage.BlockStore.BlockHash
+import coop.rchain.blockstorage.FileLMDBIndexBlockStore.Checkpoint
+import coop.rchain.blockstorage.errors.{CheckpointsAreNotConsecutive, CheckpointsDoNotStartFromZero}
+import coop.rchain.casper.protocol.BlockMessage
+import coop.rchain.shared.Resources.withResource
+import coop.rchain.blockstorage.util.byteOps._
+import coop.rchain.shared.Log
+import coop.rchain.shared.ByteStringOps._
+import org.lmdbjava.DbiFlags.MDB_CREATE
+import org.lmdbjava._
+
+import scala.collection.JavaConverters._
+import scala.ref.WeakReference
+import scala.util.control.NonFatal
+import scala.util.matching.Regex
+
+class FileLMDBIndexBlockStore[F[_]: Monad: Sync] private (
+    lock: Semaphore[F],
+    env: Env[ByteBuffer],
+    index: Dbi[ByteBuffer],
+    blockMessageRandomAccessFileRef: Ref[F, RandomAccessFile],
+    checkpointsRef: Ref[F, List[Checkpoint]]
+) extends BlockStore[F] {
+  private[this] def withTxn[R](txnThunk: => Txn[ByteBuffer])(f: Txn[ByteBuffer] => R): F[R] =
```

Comment:
this is copy-paste from `LMDBBlockStore`, copied, not reused

Reply:
The plan is to replace `LMDBBlockStore` with this one so `LMDBBlockStore` will have to go away eventually (once I merge all PRs regarding new block storage).

Reply:
Ok, in that case consider cleaning it up. Use `Log.error("...", ex)` instead of `ex.printStackTrace`